e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/foo/ExtraFitnessMongoFilterTest.java

```diff
@@ -0,0 +1,748 @@
+
+        EvaluatedIndividual ei = ff.calculateCoverage(individual);
+        FitnessValue fv = ei.getFitness();
+
+        //as no data in database meets criterion, should get distance greater than zero
+        assertTrue(fv.averageExtraDistancesToMinimize(1) > 0);
+
+    }
+
+    @Test
+    public void testFindByAgeGreaterThan() {
+
+        String[] args = new String[]{
+                "--createTests", "true",
+                "--seed", "42",
+                "--sutControllerPort", "" + controllerPort,
+                "--maxActionEvaluations", "2",
+                "--stoppingCriterion", "FITNESS_EVALUATIONS",
+                "--heuristicsForMongo", "true",
+                "--maxTestSize", "2"
+        };
+
+        Injector injector = Main.init(args);
+
+        FitnessFunction<RestIndividual> ff = injector.getInstance(Key.get(
+                new TypeLiteral<FitnessFunction<RestIndividual>>() {
+                }));
+
+        IntegerGene ageGene = new IntegerGene("age",
+                100,
+                Integer.MIN_VALUE,
+                Integer.MAX_VALUE,
+                new IntMutationUpdate(Integer.MIN_VALUE, Integer.MAX_VALUE, 0, false));
+
+        PathParam pathParam = new PathParam("age",
+                new DisruptiveGene<>(
+                        "age",
+                        ageGene,
+                        1.0d
+                )
+        );
+
+        RestCallAction postAction = new RestCallAction("POST/api/mongoperson/addJoeBlack",
+                HttpVerb.POST, new RestPath("/api/mongoperson/addJoeBlack"),
+                new LinkedList<>(),
+                new NoAuth(),
+                false,
+                null,
+                new LinkedList<>(),
+                new HashMap<>());
+
+        RestCallAction getAction = new RestCallAction("GET/api/mongoperson/findByAgeGreaterThan/{age}",
+                HttpVerb.GET,
+                new RestPath("/api/mongoperson/findByAgeGreaterThan/{age}"),
+                Arrays.asList(pathParam),
+                new NoAuth(),
+                false,
+                null,
+                new LinkedList<>(),
+                new HashMap<>());
+
+        RestIndividual individual = new RestIndividual(Arrays.asList(postAction, getAction),
+                SampleType.RANDOM,
+                new LinkedList<>(),
+                null,
+                null);
+
+        FindIterable<Document> findIterable0 = sutController.getMongoClient().getDatabase("testdb").getCollection("person").find();
+        assertFalse(findIterable0.iterator().hasNext());
+
+        EvaluatedIndividual ei = ff.calculateCoverage(individual);
+        FitnessValue fv = ei.getFitness();
+
+        //as no data in database meets criterion, should get distance greater than zero
+        assertTrue(fv.averageExtraDistancesToMinimize(1) > 0);
+
+    }
+
+    @Test
+    public void testFindByAgeLessThan() {
+
+        String[] args = new String[]{
```

Comment:
isn't this `args` repeated in different tests? i guess could be put in a private method

Reply:
ok